src/Var.jl

```diff
@@ -355,12 +352,54 @@ Return a new OutputVar where the values are averaged arithmetically in time.
 """
 function average_time(var; ignore_nan = true)
     reduced_var = _reduce_over(ignore_nan ? nanmean : mean, time_name(var), var)
+    _update_long_name_average!(reduced_var, var, time_name(var))
+    return reduced_var
+end
+
+"""
+    dim_units(var::OutputVar, dim_name)
+
+Return the `units` of the given `dim_name` in `var`, if available.
+
+If not available, return an empty string.
+"""
+function dim_units(var::OutputVar, dim_name)
+    !haskey(var.dims, dim_name) &&
+        error("Var does not have dimension $dim_name, found $(keys(var.dims))")
+    # Double get because var.dim_attributes is a dictionry whose values are dictionaries 
+    get(get(var.dim_attributes, dim_name, Dict()), "units", "")
+end
+
+"""
+    range_dim(var::OutputVar, dim_name)
+
+Return the range of the dimension `dim_name` in `var`. 
+"""
+function range_dim(var::OutputVar, dim_name)
+    !haskey(var.dims, dim_name) &&
+        error("Var does not have dimension $dim_name, found $(keys(var.dims))")
+    first_elt = first(var.dims[dim_name])
+    last_elt = last(var.dims[dim_name])
+    return first_elt, last_elt
+end
+
+"""
+    _update_long_name_average!(reduced_var::OutputVar, var::OutputVar, dim_name)
+
+Used by averaging functions to update the long name of `reduced_var` by describing what the
+average is being taken over and the associated units. 
+"""
+function _update_long_name_average!(
+    reduced_var::OutputVar,
+    var::OutputVar,
+    dim_name,
+)
+    dim_of_units = dim_units(var, dim_name)
+    first_elt, last_elt = range_dim(var, dim_name)
 
     if haskey(var.attributes, "long_name")
-        reduced_var.attributes["long_name"] *= " averaged over time"
+        reduced_var.attributes["long_name"] *= " averaged over $dim_name ($first_elt to $last_elt$dim_of_units)"
     end
```

Comment:
```suggestion
    end
    return nothing
```
Adding a `return nothing` ensures that the function has a predictable and stable return type.